### PR TITLE
fix: Feedback - button and crashes

### LINKED ISSUE #2175: UI overlapping in event details
Feedback button is overlapping with tickets on bull bottom scroll if there are no similar events.

 

**Would you like to work on the issue?**

yes

### LINKED ISSUE #2184: App crashes on writing feedback after 2nd
**Logs**
```
java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid view holder adapter positionViewHolder{4bc42fd position=2 id=-1, oldPos=1, pLpos:1 scrap [attachedScrap] tmpDetached no parent} androidx.recyclerview.widget.RecyclerView{96bbde4 VFED..... ......I. 0,202-1080,566 #7f0901ad app:id/feedbackRv}, adapter:org.fossasia.openevent.general.feedback.FeedbackRecyclerAdapter@1d608aa, layout:androidx.recyclerview.widget.LinearLayoutManager@c41ac9b, context:org.fossasia.openevent.general.MainActivity@8544568
        at androidx.recyclerview.widget.RecyclerView$Recycler.validateViewHolderForOffsetPosition(RecyclerView.java:5953)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6137)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6097)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6093)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
```

**Would you like to work on the issue?**

yes

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #2175 
Fixes #2184

<p align="center">
 
</p>

### CODE REVIEW

File: app/src/main/res/layout/content_event.xml [hunk 1]
```diff
@@ -555,14 +555,14 @@
                     android:visibility="gone"
                     tools:visibility="visible"/>
 
-                <Button
-                    android:id="@+id/feedbackBtn"
+                <TextView
```
Comment (iamareebjamal) [inline app/src/main/res/layout/content_event.xml]: Why?

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: TextView is looking good instead of the button same as tickets

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: But it is not a Button, also it is horrible for accesability. Secondly, it won't be selected as a button on navigration through keyboard

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: But why it requires navigation through the keyboard. It is displaying an alert dialog to write feedback. Then the dialog button will submit the feedback.

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: Everything should be accessible. If you're changing a button to textview, why not do it for all layouts? Let's open a merge request in AOSP to remove Button class. Let's deprecate it in favor of TextView. Why do we even need button when there is TextView?

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: https://material.io/design/usability/accessibility.html#implementing-accessibility

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: https://developer.android.com/guide/topics/ui/accessibility/apps.html#navigation

Comment (anhanh11001) [reply app/src/main/res/layout/content_event.xml]: You can try
`            style="@style/Widget.MaterialComponents.Button.TextButton"
`
@liveHarshit

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: It needs to be a Button

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: Agree with you. There are other text views also using for click listeners. Should I replace them also?

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: Yes

**Comment by liveHarshit** [comment]
Updated

File: app/src/main/res/layout/content_event.xml [hunk 2]
```diff
@@ -555,14 +555,15 @@
                     android:visibility="gone"
                     tools:visibility="visible"/>
 
-                <Button
+                <com.google.android.material.button.MaterialButton
                     android:id="@+id/feedbackBtn"
+                    style="@style/ThreeDS2TextButtonStyle"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/content_event.xml]: What's this?

File: app/src/main/res/layout/content_event.xml [hunk 3]
```diff
@@ -555,14 +555,15 @@
                     android:visibility="gone"
                     tools:visibility="visible"/>
 
-                <Button
+                <com.google.android.material.button.MaterialButton
```
Comment (iamareebjamal) [inline app/src/main/res/layout/content_event.xml]: Why? It happens automatically when android inflates the layout

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -356,11 +356,11 @@
     <string name="and_i_m_up_for">And I\'m up for</string>
     <string name="what_type">What Sounds good?</string>
     <string name="feedback">Feedback</string>
-    <string name="write_feedback">Write Feedback</string>
+    <string name="write_feedback">WRITE A FEEDBACK</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: Revert

Comment (liveHarshit) [reply app/src/main/res/values/strings.xml]: Okay. I just follow the play store for UI.

File: app/src/main/res/layout/content_event.xml (hunk 2 above)
Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: For UI.

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: What does it do which is not in default style?

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/res/values/strings.xml]: Button text is automatically capitalized

Comment (liveHarshit) [reply app/src/main/res/values/strings.xml]: Yes right sorry. I have changed it for text view.

File: app/src/main/res/layout/content_event.xml (hunk 2 above)
Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: It shows elevation. Even if a transparent background is set.

**Comment by liveHarshit** [comment]
Updated

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (liveHarshit) [reply app/src/main/res/values/strings.xml]: Not capitalizing with that style. Added an extra attribute.

File: app/src/main/res/layout/content_event.xml [hunk 5]
```diff
@@ -557,12 +557,14 @@
 
                 <Button
                     android:id="@+id/feedbackBtn"
+                    style="@style/ThreeDS2TextButtonStyle"
+                    android:textAllCaps="true"
+                    android:textAlignment="center"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/content_event.xml]: Unneeded

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: removed

File: app/src/main/res/layout/content_event.xml (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: Merging this. But I see no documentation for this

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: Found it. It is from the stripe library.

Comment (iamareebjamal) [reply app/src/main/res/layout/content_event.xml]: Wow. Please revert it in the next development iteration ASAP.

Comment (liveHarshit) [reply app/src/main/res/layout/content_event.xml]: > Wow. Please revert it in the next development iteration ASAP.

Sure :+1:

**Pull request merged**